[bug] Cannot find module '@wagmi/core/src/actions/accounts/signTypedData' or its corresponding type declarations

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.4.12

### Current Behavior

When trying to compile a custom connector for wagmi I get this error:
```
$ yarn rimraf dist && tsc
$ /Users/mmv/projects/gnosis/safe-apps-sdk/node_modules/.bin/rimraf dist
../../node_modules/wagmi/dist/declarations/src/hooks/accounts/useSignTypedData.d.ts:6:21 - error TS2307: Cannot find module '@wagmi/core/src/actions/accounts/signTypedData' or its corresponding type declarations.

6     domain?: import("@wagmi/core/src/actions/accounts/signTypedData").TypedDataDomain | undefined;
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../node_modules/wagmi/dist/declarations/src/hooks/accounts/useSignTypedData.d.ts:7:35 - error TS2307: Cannot find module '@wagmi/core/src/actions/accounts/signTypedData' or its corresponding type declarations.

7     types?: Record<string, import("@wagmi/core/src/actions/accounts/signTypedData").TypedDataField[]> | undefined;

```

It seems that the npm package imports types from the source folder of wagmi, which are not included.

I tried to compile wagmi myself and I can confirm that the file `wagmi/packages/react/dist/declarations/src/hooks/accounts/useSignTypedData.d.ts` contains an import from the `src` folder

### Expected Behavior

All type declarations are included in the bundle pushed to npm

### Steps To Reproduce

1. clone wagmi repo
2. pnpm install
3. pnpm build
4. go to `wagmi/packages/react/dist/declarations/src/hooks/accounts/useSignTypedData.d.ts` and observe

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

_No response_

### Anything else?

_No response_

--- a/.changeset/poor-carpets-repeat.md
+++ b/.changeset/poor-carpets-repeat.md
@@ -1,0 +1,6 @@
+---
+'@wagmi/core': minor
+'wagmi': minor
+---
+
+**Breaking**: `TypedDataDomain` and `TypedDataField` types were removed and incorporated into `SignTypedDataArgs`.

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -6,9 +6,20 @@
   "repository": "tmm/wagmi",
   "author": "awkweb.eth",
   "ethereum": "awkweb.eth",
+  "funding": [
+    {
+      "type": "gitcoin",
+      "url": "https://gitcoin.co/grants/4493/wagmi-react-hooks-library-for-ethereum"
+    },
+    {
+      "type": "github",
+      "url": "https://github.com/sponsors/tmm"
+    }
+  ],
   "sideEffects": false,
   "main": "dist/wagmi-core.cjs.js",
   "module": "dist/wagmi-core.esm.js",
+  "types": "dist/wagmi-core.cjs.d.ts",
   "exports": {
     "./package.json": "./package.json",
     ".": {

--- a/packages/core/src/actions/accounts/signTypedData.ts
+++ b/packages/core/src/actions/accounts/signTypedData.ts
@@ -7,24 +7,17 @@ import {
 } from '../../errors'
 import { fetchSigner } from './fetchSigner'
 
-export interface TypedDataDomain {
-  name?: string
-  version?: string
-  chainId?: BigNumberish
-  verifyingContract?: string
-  salt?: BytesLike
-}
-
-export interface TypedDataField {
-  name: string
-  type: string
-}
-
 export type SignTypedDataArgs = {
   /** Domain or domain signature for origin or contract */
-  domain: TypedDataDomain
+  domain: {
+    name?: string
+    version?: string
+    chainId?: BigNumberish
+    verifyingContract?: string
+    salt?: BytesLike
+  }
   /** Named list of all type definitions */
-  types: Record<string, Array<TypedDataField>>
+  types: Record<string, Array<{ name: string; type: string }>>
   /** Data to sign */
   value: Record<string, any>
 }

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -20,6 +20,7 @@
   "sideEffects": false,
   "main": "dist/wagmi.cjs.js",
   "module": "dist/wagmi.esm.js",
+  "types": "dist/wagmi.cjs.d.ts",
   "exports": {
     "./package.json": "./package.json",
     ".": {

--- a/packages/react/src/hooks/accounts/useConnect.ts
+++ b/packages/react/src/hooks/accounts/useConnect.ts
@@ -24,9 +24,8 @@ export type UseConnectConfig = {
   onSettled?: MutationOptions['onSettled']
 }
 
-export const mutationKey = (args: UseConnectArgs) => [
-  { entity: 'connect', ...args },
-]
+export const mutationKey = (args: UseConnectArgs) =>
+  [{ entity: 'connect', ...args }] as const
 
 const mutationFn = (args: UseConnectArgs) => {
   const { connector, chainId } = args

--- a/packages/react/src/hooks/accounts/useDisconnect.ts
+++ b/packages/react/src/hooks/accounts/useDisconnect.ts
@@ -17,7 +17,7 @@ export type UseDisconnectConfig = {
   onSuccess?: (context: unknown) => void | Promise<unknown>
 }
 
-export const mutationKey = [{ entity: 'disconnect' }]
+export const mutationKey = [{ entity: 'disconnect' }] as const
 
 const mutationFn = () => disconnect()
 

--- a/packages/react/src/hooks/accounts/useNetwork.ts
+++ b/packages/react/src/hooks/accounts/useNetwork.ts
@@ -20,9 +20,8 @@ export type UseNetworkConfig = MutationConfig<
   SwitchNetworkArgs
 >
 
-export const mutationKey = (args: UseNetworkArgs) => [
-  { entity: 'switchNetwork', ...args },
-]
+export const mutationKey = (args: UseNetworkArgs) =>
+  [{ entity: 'switchNetwork', ...args }] as const
 
 const mutationFn = (args: UseNetworkArgs) => {
   const { chainId } = args

--- a/packages/react/src/hooks/accounts/useSignMessage.ts
+++ b/packages/react/src/hooks/accounts/useSignMessage.ts
@@ -12,9 +12,8 @@ export type UseSignMessageConfig = MutationConfig<
   SignMessageArgs
 >
 
-export const mutationKey = (args: UseSignMessageArgs) => [
-  { entity: 'signMessage', ...args },
-]
+export const mutationKey = (args: UseSignMessageArgs) =>
+  [{ entity: 'signMessage', ...args }] as const
 
 const mutationFn = (args: UseSignMessageArgs) => {
   const { message } = args

--- a/packages/react/src/hooks/accounts/useSignTypedData.ts
+++ b/packages/react/src/hooks/accounts/useSignTypedData.ts
@@ -16,9 +16,8 @@ export type UseSignTypedDataConfig = MutationConfig<
   SignTypedDataArgs
 >
 
-export const mutationKey = (args: UseSignTypedDataArgs) => [
-  { entity: 'signTypedData', ...args },
-]
+export const mutationKey = (args: UseSignTypedDataArgs) =>
+  [{ entity: 'signTypedData', ...args }] as const
 
 const mutationFn = (args: UseSignTypedDataArgs) => {
   const { domain, types, value } = args

